website/docs/r/nsxt_dynamic_security_group.html.markdown

```diff
@@ -0,0 +1,196 @@
+}
+
+resource "vcd_nsxt_dynamic_security_group" "group1" {
+  org          = "cloud"
+  vdc_group_id = data.vcd_vdc_group.group1.id
+
+  name = "dynamic-security-group-example"
+
+  criteria { # Boolean "OR"
+    rule {   # Boolean "AND"
+      type     = "VM_TAG"
+      operator = "EQUALS"
+      value    = "tag-equals"
+    }
+
+    rule { # Boolean "AND"
+      type     = "VM_TAG"
+      operator = "CONTAINS"
+      value    = "tag-contains"
+    }
+
+    rule { # Boolean "AND"
+      type     = "VM_TAG"
+      operator = "STARTS_WITH"
+      value    = "tag-starts-with"
+    }
+
+    rule { # Boolean "AND"
+      type     = "VM_TAG"
+      operator = "ENDS_WITH"
+      value    = "tag-ends-with"
+    }
+  }
+
+  criteria { # Boolean "OR" evaluation
+    rule {   # Boolean "AND"
+      type     = "VM_NAME"
+      operator = "CONTAINS"
+      value    = "name-contains-4"
+    }
+
+    rule { # Boolean "AND"
+      type     = "VM_NAME"
+      operator = "STARTS_WITH"
+      value    = "starts_with2"
+    }
+
+    rule { # Boolean "AND"
+      type     = "VM_NAME"
+      operator = "CONTAINS"
+      value    = "name-contains-22"
+    }
+
+    rule { # Boolean "AND"
+      type     = "VM_NAME"
+      operator = "STARTS_WITH"
+      value    = "starts_with22"
+    }
+  }
+
+  criteria { # Boolean "OR" evaluation
+    rule {   # Boolean "AND"
+      type     = "VM_NAME"
+      operator = "CONTAINS"
+      value    = "name-contains3"
+    }
+
+    rule { # Boolean "AND"
+      type     = "VM_NAME"
+      operator = "STARTS_WITH"
+      value    = "starts_with3"
+    }
+
+    rule { # Boolean "AND"
+      type     = "VM_NAME"
+      operator = "CONTAINS"
+      value    = "name-contains33"
+    }
+
+    rule { # Boolean "AND"
+      type     = "VM_NAME"
+      operator = "STARTS_WITH"
+      value    = "starts_with33"
+    }
+  }
+}
+```
+
+## Example Usage 2 (Empty Dynamic Security Group)
+```hcl
+data "vcd_vdc_group" "group1" {
+  org  = "cloud"
+  name = "vdc-group-cloud"
+}
+
+resource "vcd_nsxt_dynamic_security_group" "group1" {
+  org          = "cloud"
+  vdc_group_id = data.vcd_vdc_group.group1.id
+
+  name = "empty-dynamic-security-group"
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `org` - (Optional) The name of organization to use, optional if defined at provider level. Useful
+  when connected as sysadmin working across different organisations.
+* `vdc_group_id` - (Required) VDC Group ID for Dynamic Security Group creation.
+* `name` - (Required) A unique name for Dynamic Security Group
+* `description` - (Optional) An optional description of the Dynamic Security Group
+* `criteria` (Optional) Up to 3 criteria for matching VMs. List of criteria is matched with boolean
+  `OR` operation and matching any of defined criteria will include objects. Each `criteria` can
+  contains up to 4 `rule` definitions.
+* `rule` (Optional) Up to 4 rules for matching VMs. List of rules are matched with boolean `AND`
```

Comment:
Two more places here where maybe we should state that 3 and 4 limits are as of specific VCD version? I'm suggesting this because you wrote that it may become higher in future versions.

Reply:
Got it. Then let's not spread the doubt :)